src/obb/server/http_component.clj

```diff
@@ -15,19 +15,21 @@
                   [#"^http://localhost(.*)"]
                   :access-control-allow-methods [:get :put :post :delete]))
 
-(def app
+(defn app
   "The main app handler"
+  [system]
   (-> (compojure/routes routes/public-routes)
       (setup-cors)))
 
-(defrecord HttpServerComponent [server port]
+(defrecord HttpServerComponent [server port meta]
 
   component/Lifecycle
 
   (start [component]
     (println (str "** OBB HTTP Server " (or (get (System/getenv) "OBB_ENV" "development"))
                   " running on port " port))
-    (assoc component :server (http/start-server app {:port port})))
+    (assoc component :server (http/start-server (app (:system meta))
+                                                     {:port port})))
```

Comment:
We found this information which is what we decided to do.

We will use a system to declare all the dependencies between components and a component that uses all the required components for a web request to be served.
The system is started at application boot and the general component is added to the web request in a ring middleware. Refer to [this link](https://github.com/stuartsierra/component#entry-points-in-production) for more information.